src/scgpt/embedding/script.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+import mudata as mu
+import json
+from scgpt.tokenizer.gene_tokenizer import GeneVocab
+from scgpt.model import TransformerModel
+from scgpt.utils.util import load_pretrained
+import torch
+
+## VIASH START
+par = {
+    "input": "resources_test/scgpt/test_resources/Kim2020_Lung_tokenized.h5mu",
+    "obsm_gene_tokens": 'gene_id_tokens',
+    "obsm_tokenized_values": 'values_tokenized',
+    "obsm_padding_mask": 'padding_mask',
+    "model": "resources_test/scgpt/source/best_model.pt",
+    "model_config": "resources_test/scgpt/source/args.json",
+    "model_vocab": "resources_test/scgpt/source/vocab.json",
+    "output": "Kim2020_Lung_embedded.h5ad",
+    "var_gene_names": "gene_name",
+    "obs_batch_label": "sample",
+    "obsm_embeddings": "X_scGPT",
+    "pad_token": "<pad>",
+    "pad_value": -2,
+    "batch_size": 64,
+    "modality": "rna",
+    "dropout": 0.2,
+    "DSBN": True,
+    "n_input_bins": 51,
+}
+## VIASH END
+
+# START TEMPORARY WORKAROUND setup_logger
+# reason: resources aren't available when using Nextflow fusion
+# from setup_logger import setup_logger
+def setup_logger():
+    import logging
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+logger = setup_logger()
+
+
+#TODO: Optionally set device to use gpu, to be implemented when gpu-based machine types are supported
```

Comment:
Supported by who? scGPT or one of the compute environments?

Reply:
Compute environments. Currently we can't run it yet using batch, so as soon as this is possible infrastructurally we can uncomment this code